src/libcpu/src/jit/jit.cpp

```diff
@@ -265,16 +379,25 @@ gen(JitBlock &block)
          a.bind(targetIter->second.label);
       }
 
-      if (JIT_DEBUG) {
```

Comment:
I should mention it's important that core->nia remain unchanged until after the pre-verify call, so it's not already updated for the Core instance the interpreter sees (which could potentially mask bugs).